### PR TITLE
Unit 4 PSE 2 C26 Updates

### PR DESCRIPTION
**Pull request by mikellewade**
[Asana Ticket](https://app.asana.com/1/181459410160484/project/1205655776549324/task/1215427716613232?focus=true)
[Core PSE PRs](https://github.com/Ada-Developers-Academy/core-pses/pulls?q=is%3Apr+) in case it's useful to compare changes against one of those.

### CODE REVIEW

File: 02/implement-intersection.checkpoint.md [hunk 1]
```diff
@@ -583,10 +585,12 @@ What is the time complexity of your solution? Please define and explain your var
 * points: 1 
 ##### !question
 
-What is the space complexity of your solution? Please define and explain your variables.
+What is the space complexity of your solution? xplain your reasoning. Define your variable(s) (i.e. n is the length of the list).
```
Comment (kelsey-steven-ada) [inline 02/implement-intersection.checkpoint.md]:
```suggestion
What is the space complexity of your solution? Explain your reasoning. Define your variable(s) (i.e. n is the length of the list).
```

File: 02/intersection.md [hunk 2]
```diff
@@ -205,6 +135,8 @@ The reviewer pointed out additional useful observations, like the fact that the
     * What is the expected output for the given input?
     * You can use the examples provided in the prompt, or other examples.
 2. Write unit tests for `find_intersecting_node` for the nominal and edge cases you identified in the first step.
+When naming a test, we want to ensure the name describes the scenario we are testing by including information like the function being tested, inputs, and expected outputs.
```
Comment (kelsey-steven-ada) [inline 02/intersection.md:139]: Needs the spacing above or the note gets tacked on to the "2." paragraph.

```suggestion
2. Write unit tests for `find_intersecting_node` for the nominal and edge cases you identified in the first step.

When naming a test, we want to ensure the name describes the scenario we are testing by including information like the function being tested, inputs, and expected outputs. 
```

File: 02/intersection.md [hunk 3]
```diff
@@ -325,7 +257,7 @@ As you write your steps, keep the following guidelines in mind:
 * The steps should be a description as if you were talking out the problem with another person and should be agnostic of any particular language. 
     * As such, they should not include code syntax in the description.
 
-What's important at this stage is to think through and outline the implementation before writing code.
+What's important at this stage is to think through ahow you would put an approach into words, and outline the implementation before writing code.
```
Comment (kelsey-steven-ada) [inline 02/intersection.md]:
```suggestion
What's important at this stage is to think through how you would put an approach into words, and outline the implementation before writing code.
```

File: 02/intersection.md [hunk 4]
```diff
@@ -402,7 +343,21 @@ Chat Link: `<url to your conversation>`
...
+
+<br>
+
+One suggestion that stood out to me was the recommendation to mention the classic technique of aligning the two lists before comparing them, rather than comparing every node of list 1 against every node of list 2. The AI explained that walking each list once to find its length, then advancing the pointer of the longer list by the length difference before stepping through both lists together, would let me find the intersection in a single pass instead of a nested one. This shifted my understanding of the roadmap goal versus the communication goal, since my nested-loop steps are correct and easy to explain, but naming this alternative up front shows an interviewer that I recognize a brute-force solution's limitations even if I choose to implement the simpler version first.
+
+<br>
```
Comment (kelsey-steven-ada) [inline 02/intersection.md]: Parts of the new explanations felt a little weird like the paragraphs starting very similarly and some sections repeating points in themselves. I made a pass at trimming the paragraphs down a little bit, let me know what you think. 

```suggestion
The AI tool pointed out that my steps compare nodes by identity rather than by value and noted that this detail made sure the roadmap solves the stated problem (since the prompt distinguishes true intersections from lists that share equal values). I think this was effective because it shows I broke down the prompt and pulled specific requirements out over assuming we're trying to solve the "equal values" version of this problem.

<br>

One improvement the AI recommended was to make the stopping condition of each loop explicit rather than implied by steps like "iterate through the nodes of list 1". The AI pointed out that in a spoken interview, leaving this implicit can make it sound like I haven't thought through the termination condition, even though I likely have. If I were rewriting my steps, I would state that each loop advances its pointer until it reaches a node whose next value is empty, at which point the loop ends. 

<br>

A piece of feedback I believe was inaccurate was the claim that resetting the list 2 pointer back to the head on every pass through the outer loop was a mistake that would hurt performance and should be removed. Resetting the pointer is required for the nested-loop approach to work; removing the reset would mean most nodes of list 1 never get compared against most nodes of list 2, causing valid intersections to be missed. A valid concern worth raising is the resulting time complexity, not the reset itself, so this feedback addressed a problem that does not exist in the intended approach.

<br>

There was a recommendation that stood out to me around mentioning the technique of aligning the two lists before comparing them, rather than comparing every node of list 1 against every node of list 2. The AI explained that walking each list once to find its length, then advancing the pointer of the longer list by the length difference before stepping through both lists together, would let me find the intersection in a single pass instead of a nested one. This shifted my understanding of the roadmap goal versus the communication goal, since my nested-loop steps are correct and easy to explain, but naming this alternative up front shows an interviewer that I recognize a brute-force solution's limitations even if I choose to implement the simpler version first.

<br>
```

**Review by kelsey-steven-ada — CHANGES_REQUESTED** [review]
Looking good overall, I made a couple suggestions & there are a few typos to address.

**Review by kelsey-steven-ada — APPROVED** [review]
LGTM!
